lib/internal/worker.js

```diff
@@ -415,6 +426,50 @@ function makeResourceLimits(float64arr) {
   };
 }
 
+function eventLoopUtilization(util1, util2) {
+  // TODO(trevnorris): Works to solve the thread-safe read/write issue of
```

Comment:
Would it make sense for the API instead to be emitted on `online` so:

 - The worker emits `online` with an "stats" argument
 - The stats argument has this `eventLoopUtilization` method the user can call

That would make it impossible to use this API incorrectly and remove a bit of the weirdness.

Reply:
That seems like it makes accessing the API more difficult, with little benefit…

Reply:
@addaleax
Well, the benefit is that it would be impossible to misuse the API. Currently this API has a (relatively small) footgun if you call `eventLoopUtilization` before `online` is fired. That eliminates the footgun.

Note that the API isn't really that bad:

```js
const worker = new Worker(...);
const controller = await once(worker, 'online');
console.log(controller.eventLoopUtilization()); // same thing as currently on worker now
```

Reply:
@benjamingr I think my disagreement is more about whether this is a footgun or not :) If you call it before the `'online'` event, you don’t get inaccurate data, you get no ELU data – which is *accurate*, because (up to race conditions) there actually is no ELU data available.

Reply:
If you think the distinction between "no data" and "0 data" here isn't important - I will concede.

Reply:
@benjamingr I plan on figuring out a way to remove the limitation of not being able to query the ELU until the `'online'` event has been emitted.